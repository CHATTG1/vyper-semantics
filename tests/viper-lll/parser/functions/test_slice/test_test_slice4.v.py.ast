%pgm(, ,
  %fdecl(%@public, foo, %param(inp, %bytesT(10)) %param(start, %num) %param(len, %num), %bytesT(10),
    %return(%slice(%var(inp), %var(start), %var(len))))
)
