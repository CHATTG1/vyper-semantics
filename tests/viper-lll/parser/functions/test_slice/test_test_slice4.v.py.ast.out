$%seq ( $return ( 0 , $lll ( $%seq ( $%seq ( $mstore ( 28 , $calldataload ( 0 ) ) , $mstore ( 32 , 1461501637330902918203684832716283019655932542976 ) , $mstore ( 64 , 170141183460469231731687303715884105727 ) , $mstore ( 96 , -170141183460469231731687303715884105728 ) , $mstore ( 128 , 1701411834604692317316873037158841057270000000000 ) , $mstore ( 160 , -1701411834604692317316873037158841057280000000000 ) ) , $if ( $eq ( $mload ( 0 ) , 2223042402 ) , $%seq ( $calldatacopy ( 320 , 4 , 96 ) , $assert ( $iszero ( $callvalue ) ) , $%seq ( $calldatacopy ( 416 , $add ( 4 , $calldataload ( $add ( 4 , 0 ) ) ) , 42 ) , $assert ( $le ( $calldataload ( $add ( 4 , $calldataload ( $add ( 4 , 0 ) ) ) ) , 10 ) ) ) , $clamp ( $mload ( 96 ) , $calldataload ( $add ( 4 , 32 ) ) , $mload ( 64 ) ) , $clamp ( $mload ( 96 ) , $calldataload ( $add ( 4 , 64 ) ) , $mload ( 64 ) ) , $with ( $_loc , $with ( $_start , $mload ( 352 ) , $with ( $_length , $mload ( 384 ) , $with ( $_opos , $add ( 512 , $mod ( $_start , 32 ) ) , $%seq ( $assert ( $lt ( $add ( $_start , $_length ) , 10 ) ) , $with ( $_l , 10 , $pop ( $call ( 19 , 4 , 0 , $add ( 416 , $add ( $sub ( $_start , $mod ( $_start , 32 ) ) , 32 ) ) , $_l , 544 , $_l ) ) ) , $mstore ( $_opos , $_length ) , $_opos ) ) ) ) , $%seq ( $mstore ( $sub ( $_loc , 32 ) , 32 ) , $return ( $sub ( $_loc , 32 ) , $ceil32 ( $add ( $mload ( $_loc ) , 64 ) ) ) ) ) , $stop ) ) ) , 0 ) ) )
